src/Plugin/migrate/process/ApplyFilters.php

```diff
@@ -0,0 +1,316 @@
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    if (!is_string($value)) {
+      throw new MigrateException(sprintf('%s is not a string.', var_export($value, TRUE)));
+    }
+    $this->row = $row;
+
+    return $this->doFilter($value);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationKeys(array $keys = NULL): void {
+    $this->validateConfigurationKeyFilterFormat('filter_format');
+    $this->validateConfigurationKeyFiltersAndFiltersToSkip(['filters_to_apply', 'filters_to_skip']);
+  }
+
+  /**
+   * Validates the "filter_format" configuration key.
+   *
+   * @param string $key
+   *   The configuration key to validate.
+   */
+  protected function validateConfigurationKeyFilterFormat(string $key): void {
+    if (!array_key_exists($key, $this->configuration) || !is_string($this->configuration[$key])) {
+      throw new \InvalidArgumentException(sprintf('The configuration key "%s" is required and must be a string.', $key));
+    }
+    if (is_null($this->getConfigurationFilterFormat())) {
+      throw new \InvalidArgumentException(sprintf('The configuration key "%s" (%s) is not a valid filter format.', $key, $this->configuration[$key]));
+    }
+  }
+
+  /**
+   * Validates the "filters_to_apply" and "filters_to_skip" configuration keys.
+   *
+   * @param string[] $keys
+   *   The configuration keys to validate.
+   */
+  protected function validateConfigurationKeyFiltersAndFiltersToSkip(array $keys): void {
+    $filter_format = $this->getConfigurationFilterFormat();
+    foreach ($keys as $key) {
+      if (array_key_exists($key, $this->configuration)) {
+        if (!is_array($this->configuration[$key]) || empty($this->configuration[$key])) {
+          throw new \InvalidArgumentException(sprintf('The configuration option "%s" must be a non-empty array.', $key));
+        }
+        else {
+          foreach ($this->configuration[$key] as $filter_id) {
+            if (!is_string($filter_id)) {
+              throw new \InvalidArgumentException(sprintf('The configuration option "%s" must contain strings. "%s" is not a string.', $key, var_export($filter_id, TRUE)));
+            }
+            if ($this->filterFormatManager->isValidFilterId($filter_id, $filter_format) !== TRUE) {
+              throw new \InvalidArgumentException(sprintf('"%s" is not a valid filter for the "%s" filter format.', $filter_id, $filter_format->id()));
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Returns the current filter format.
+   *
+   * @return \Drupal\filter\Entity\FilterFormat|null
+   *   The current filter format.
+   */
+  protected function getConfigurationFilterFormat(): ?FilterFormat {
+    return is_string($this->configuration['filter_format']) ? $this->filterFormatManager->getFilterFormat($this->configuration['filter_format']) : NULL;
```

Comment:
In `validateConfigurationKeys` you validate that we have a `filter_format` so this method can be sure that we have it no? So is there a need to return NULL ever?

Reply:
**Status:** Fixed
This is also true. It's just that the notion of a configuration validator came afterwards.
I refactored the method to become a simple getter.